### PR TITLE
Soft coordinator shutdown

### PR DESCRIPTION
**Pull request by neunhoef**
This adds the soft shutdown feature for coordinators for the 3.8
branch (backport from devel).

Here is the design document:

https://github.com/arangodb/documents/blob/master/DesignDocuments/03_IN_PROGRESS/SoftShutdownCoordinators.md

The basic idea is to allow to put a coordinator into a "soft shutdown
mode" in which it finishes ongoing work but does no longer accept
new work. This should make restarting or upgrading coordinators less
intrusive for clients.

**To be merged for 3.8.1**, together with https://github.com/arangodb/arangodb/pull/14435

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,18 @@
+v3.8.1 (XXXX-XX-XX)
```
Comment (jsteemann) [inline CHANGELOG:1]:
```suggestion
v3.8.1 (XXXX-XX-XX)
-------------------
```

File: arangod/RestServer/SoftShutdownFeature.h [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Max Neunhoeffer
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
```
Comment (jsteemann) [inline arangod/RestServer/SoftShutdownFeature.h:23]: We have moved to `#pragma once` only in devel.
I think we should use an old-style include guard in 3.8 and before.

File: arangod/RestServer/SoftShutdownFeature.h [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+#include "ApplicationFeatures/ApplicationFeature.h"
+#include "Basics/debugging.h"
+#include "Scheduler/Scheduler.h"
+#include "RestServer/Metrics.h"
```
Comment (jsteemann) [inline arangod/RestServer/SoftShutdownFeature.h:28]: I think `RestServer/Metrics.h` is not needed by this file, so we can remove the include.
```suggestion
```

File: arangod/RestServer/SoftShutdownFeature.h (hunk 2 above)
Comment (neunhoef) [reply arangod/RestServer/SoftShutdownFeature.h]: Done.

**Pull request merged**